packages/components/src/Radio/RadioButton.tsx

```diff
@@ -12,16 +12,20 @@ import * as React from 'react'
 import styled from 'styled-components'
 import { Icon } from '../Icon'
 
-const Wrapper = styled.div`
+const Label = styled.label<{ disabled?: boolean }>`
   list-style-type: none;
   display: flex;
   flex-direction: row;
   gap: 16px;
   border-radius: 4px;
   width: 100%;
   padding: 8px 8px;
-  align-items: flex-start;
-
+  align-items: center;
+  isolation: isolate;
```

Comment:
This create a stacking context thereby isolating the z-index property inside this component. You can think of this like a group that all move together & other elements won't intertwine with it even if they use similar z-index values